storagemarket/impl/client.go

```diff
@@ -209,7 +209,7 @@ func (c *Client) Start(ctx context.Context, p ClientDealProposal) (cid.Cid, erro
 		ClientCollateral:     big.Zero(),
 	}
 
-	if err := c.node.EnsureFunds(ctx, p.Client, dealProposal.ClientBalanceRequirement()); err != nil {
+	if err := c.node.EnsureFunds(ctx, p.Client, p.Client, dealProposal.ClientBalanceRequirement()); err != nil {
```

Comment:
Just curious what is the value of StorageClientNode taking two parameters if it's always going to be the same (it's clear why for the provider where the values are different)

Reply:
ah cause we have a common method in the integration test... I wonder if it would be better to just rewrite the test to not use fake common for that method.

Reply:
cause in the implementation it's actually two different entire classes that don't share anything

Reply:
(as mentioned in slack) lotus also uses a common thingy for implementing this. We *could* have them be separate interfaces, but i think the mental overhead is lower if we just have it be the same. Plus, theres no reason you couldnt fund your deal collateral with a different account